library/core/src/main/java/com/google/android/exoplayer2/extractor/ts/DvbSubtitleReader.java

```diff
@@ -30,22 +32,49 @@
  */
 public final class DvbSubtitleReader implements ElementaryStreamReader {
 
-  private final String language;
-  private final List<byte[]> initializationData;
+  private class SubtitleTrack {
+    private final String language;
+    private final List<byte[]> initializationData;
+
+    SubtitleTrack (String language, List<byte[]> initializationData) {
+      this.language = language;
+      this.initializationData = initializationData;
+    }
+  }
 
-  private TrackOutput output;
+  private List<SubtitleTrack> subtitles = new ArrayList<>();
+  private List<TrackOutput> outputTracks = new ArrayList<>();
   private boolean writingSample;
   private int bytesToCheck;
   private int sampleBytesWritten;
   private long sampleTimeUs;
 
   /**
-   * @param language The subtitle language code.
-   * @param initializationData Initialization data to be included in the track {@link Format}.
+   * @param esInfo The Elementary Stream info
    */
-  public DvbSubtitleReader(String language, byte[] initializationData) {
-    this.language = language;
-    this.initializationData = Collections.singletonList(initializationData);
+  public DvbSubtitleReader(TsPayloadReader.EsInfo esInfo) {
+    String language;
+    List<byte[]> initializationData;
+
+    int pos = 2;
+
+    while (pos + 8 <= esInfo.descriptorBytes.length) {
```

Comment:
This blindly assumes `EsInfo.descriptorBytes` only contains DVB descriptor tags. I'm not sure that's valid.

In any case, I think `TsExtractor readEsInfo` is probably the right place to be doing this parsing, given it's checking the tag type and doing some parsing already. Can we just change that method to build the list of DvbSubtitleTrack objects, each containing language + init-data, rather than just init-data that it sets currently. Then pass that list through EsInfo and I think everything is done?